### PR TITLE
feat: `gather_every` methods

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

Just something I was planning to add :)

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

**Review by MarcoGorelli — COMMENTED** [review]
thanks @FBruzzesi - I think Polars has this one for LazyFrame, should we add it there too?

**Comment by FBruzzesi** [comment]
> thanks @FBruzzesi - I think Polars has this one for LazyFrame, should we add it there too?

Apparently I can't even search in the docs 🙈 Let me fix it

File: narwhals/expr.py [hunk 1]
```diff
@@ -1726,6 +1726,49 @@ def len(self) -> Self:
...
+            >>> data = {"a": [1, 2, 3, 4], "b": [5, 6, 7, 8]}
+            >>> df_pd = pd.DataFrame(data)
+            >>> df_pl = pl.DataFrame(data)
+
+            Let's define a dataframe-agnostic function in which gather every 2 rows,
+            starting from a offset of 1:
+
+            >>> @nw.narwhalify
+            ... def func(df_any):
+            ...     return df_any.select(nw.col("a").gather_every(n=2, offset=1))
```
Comment (MarcoGorelli) [inline narwhals/expr.py:1750]:
```suggestion
            ... def func(df):
            ...     return df.select(nw.col("a").gather_every(n=2, offset=1))
```

File: narwhals/series.py [hunk 2]
```diff
@@ -2047,6 +2047,46 @@ def to_dummies(
...
+            >>> data = [1, 2, 3, 4]
+            >>> s_pd = pd.Series(name="a", data=data)
+            >>> s_pl = pl.Series(name="a", values=data)
+
+            Let's define a dataframe-agnostic function in which gather every 2 rows,
+            starting from a offset of 1:
+
+            >>> @nw.narwhalify
+            ... def func(s_any):
+            ...     return s_any.gather_every(n=2, offset=1)
```
Comment (MarcoGorelli) [inline narwhals/series.py:2071]:
```suggestion
            ... def func(s):
            ...     return s.gather_every(n=2, offset=1)
```

**Review by MarcoGorelli — APPROVED** [review]
thanks @FBruzzesi !

**Pull request merged**